### DISCUSSION TITLE
Changes are not reflecting for get hierarchy for collection/v4/system/update api

### DISCUSSION THREAD

**Discussion by somvitbhowmik** [Sunbird-Knowlg/Community#61] [General Discussions]
Hi Team,

I was trying to update the collection using "collection/v4/system/update" api. We found one issue. Below is our observation:

1:content read with and without mode=edit is reflecting the changes.
2:content search and composite search reflecting the changes.
3:get hierarchy mode=edit reflecting the changes. But without mode=edit changes are not reflecting for get hierarchy.

As I mention in the point no 3 that changes are not reflecting for get hierarchy. Pls look into this issue. @maheshkumargangula

cc: @pkranga

**Comment by AmiableAnil** [comment]
@somvitbhowmik Until the new hierarchy changes are not published, changes will not reflect without mode=edit. Please publish and then verify it once.

**Comment by karthik-tarento** [comment]
@AmiableAnil -- Is there any possibility to update the content meta directly without publishing ?
We need to update the meta and add information like the number of views, ratings for the content -- these details are updated by flink jobs which runs as per scheduled time. Any suggestion ?
cc: @somvitbhowmik

**Reply by AmiableAnil** [reply]
@karthik-tarento @somvitbhowmik There are no other option available to update the hierarchy without publishing it.
